From de558f5b964979b735a1a31fcfb6848ef7d8c007 Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Tue, 19 Jan 2016 17:09:35 +0530
Subject: [PATCH 2/5] [EM/PMIC_CCSM]: Fix enumeration issue post data role swap

Sometimes device is not getting enumerated post data role swap. This
is due to pmic ccsm driver not notifying EVENT_VBUS to the otg driver,
since by default is_notify_otg flag is set to false and also after
connecting and disconnecting wall charger, pmic ccsm driver not
notifying to otg as the flag set to flase. Hence enabled the
is_notify_otg flag by default, so by default all the events will be
notified to otg driver and also enabled the flag to notify otg if the
wall charger is removed.

Change-Id: Ic36239c401e6c4a36fadc231fcd1fc33bb1dd2d0
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-16390
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
Reviewed-on: https://android.intel.com:443/459742
---
 drivers/power/intel_pmic_ccsm.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/power/intel_pmic_ccsm.c b/drivers/power/intel_pmic_ccsm.c
index 8500bcc..10dab55 100644
--- a/drivers/power/intel_pmic_ccsm.c
+++ b/drivers/power/intel_pmic_ccsm.c
@@ -2008,6 +2008,12 @@ static void pmic_ccsm_process_cable_events(enum cable_type cbl_type,
 		if (chc.is_notify_otg) {
 			otg_evt = cable_state ? USB_EVENT_VBUS : USB_EVENT_NONE;
 			notify_otg = true;
+		} else if (!cable_state) {
+			/*
+			 * enable the flag to notify otg for the next events by
+			 * default, if the wall charger is removed.
+			 */
+			chc.is_notify_otg = true;
 		}
 		break;
 	case CABLE_TYPE_HOST:
@@ -2099,6 +2105,8 @@ static int pmic_chrgr_probe(struct platform_device *pdev)
 	chc.intmap = chc.pdata->intmap;
 	chc.intmap_size = chc.pdata->intmap_size;
 	chc.vbus_state = VBUS_ENABLE;
+	/* by default all the events should be notified to USB */
+	chc.is_notify_otg = true;
 
 	chc.pmic_model = get_pmic_model(pdev->name);
 	dev_info(chc.dev, "PMIC model is %d\n", chc.pmic_model);
-- 
1.9.1

